### ISSUE TITLE
A question about Dorefa-net

### ISSUE THREAD

**Issue by cctgem** [ppwwyyxx/tensorpack#390]
Hi,
I'm a little confused about equation 10 in the paper，"DoReFa-Net: Training Low Bitwidth Convolutional Neural Networks with Low Bitwidth Gradients". According to the formula, r0's derivative on r1 is 1-tanh2(r0), but in other open source such as BMXNet, where the derivative is 1, which one is correct?

Look forward to your reply and thanks for this wonderful toolbox!

**Comment by ppwwyyxx** [comment]
I don't understand your question. Could you explain "According to the formula, r0's derivative on r1 is 1-tanh2(r0)"

**Comment by cctgem** [comment]
Sorry, I didn't explain it clearly. 
My question is how to calculate the derivatives (∂ro/∂ri) in equation 10. In other words, how to calculate the derivatives (∂Wb/∂W), Wb is the weight after quantization and W is the origin float-point weight.
My result is tanh(W)'==(1-tanh(W) * tanh(W)), but i see the other result in other realization.

**Comment by ppwwyyxx** [comment]
It can be computed by standard chain rule.
